### PR TITLE
Feat/csr 858/pinned activated

### PR DESCRIPTION
**Pull request by wouterbaltus**

### CODE REVIEW

File: src/components/data-grid/DataGrid.tsx [hunk 1]
```diff
@@ -128,22 +154,37 @@ export const DataGrid = ({
...
+      }
+    }
   };
 
-  const handleCreateView = async (view: DataGridView) => {
-    onSelectView(view);
-    if (onCreateView) {
-      await onCreateView(view);
+  const handleCreateView = async (input: CreateViewInput) => {
+    if (onCreateView && onPinView && onActivateView) {
```
Comment (hiddeottens) [inline src/components/data-grid/DataGrid.tsx:177]: `onPinView && onActivateView` is unnecessary

File: src/components/data-grid/DataGrid.tsx [hunk 2]
```diff
@@ -128,22 +154,37 @@ export const DataGrid = ({
...
+
+      if (view.id && onActivateView) {
+        await onActivateView(view.id);
+      } else if (onDeactivateView) {
+        const activeView = allViews.find((view) => view.active);
+        if (activeView) {
+          await onDeactivateView(activeView.id);
+        }
+
+        view.active = true;
```
Comment (hiddeottens) [inline src/components/data-grid/DataGrid.tsx:170]: I would move

```
view.active = true;
setAllViews([...allViews.filter((x) => x.id !== id), view]);
```

 to right after:

```
setViewState(view?.viewState!);
```

to make sure there is no delay between activating a view and seeing the button change

File: src/components/data-grid/DataGrid.tsx [hunk 3]
```diff
@@ -128,22 +154,37 @@ export const DataGrid = ({
...
-    setSelectedView(view);
-    setViewState(view?.viewState!);
+  const handleActivateView = async (id: string) => {
+    const view = allViews?.find((view) => view.id === id);
+    if (view) {
+      setViewState(view?.viewState!);
+
+      if (view.id && onActivateView) {
+        await onActivateView(view.id);
+      } else if (onDeactivateView) {
```
Comment (hiddeottens) [inline src/components/data-grid/DataGrid.tsx:164]: This shouldn't be `else if` right?

You should always deactivate the `activeView`

**Review by hiddeottens — CHANGES_REQUESTED** [review]
Don't forget to bump the version number to `0.12.0`

File: src/components/data-grid/DataGrid.tsx (hunk 3 above)
Comment (wouterbaltus) [reply src/components/data-grid/DataGrid.tsx:164]: I thougt backend should deactivate the old active view if you activate a new one

File: src/components/data-grid/DataGrid.tsx (hunk 2 above)
Comment (wouterbaltus) [reply src/components/data-grid/DataGrid.tsx:170]: Fixed

File: src/components/data-grid/DataGrid.tsx (hunk 1 above)
Comment (wouterbaltus) [reply src/components/data-grid/DataGrid.tsx:177]: Fixed

File: src/components/data-grid/DataGrid.tsx (hunk 3 above)
Comment (wouterbaltus) [reply src/components/data-grid/DataGrid.tsx:162]: Discussed with slack. Is ok.

**Review by hiddeottens — APPROVED** [review]

**Pull request merged**